Using lodash-es

If you configure a lodash alias in webpack:

```
config.resolve.alias['lodash'] = path.dirname(require.resolve('lodash-es'))
```

RHL fails with:

```
Uncaught TypeError: merge is not a function
    at updateProxyById (react-hot-loader.development.js:1089)
    at Object.register (react-hot-loader.development.js:2533)
```

Because `const merge = require('lodash-es/merge')` results in `{ default: merge }`.
Haven't figured out how to workaround this yet.

I'm wondering the following error is related to this issue:
```bash

ERROR in ./node_modules/react-hot-loader/dist/react-hot-loader.development.js
Module not found: Error: Can't resolve 'lodash/merge' in '/Users/mcrunix/w/phizog/node_modules/react-hot-loader/dist'
 @ ./node_modules/react-hot-loader/dist/react-hot-loader.development.js 1057:12-35
 @ ./node_modules/react-hot-loader/patch.js
 @ multi react-hot-loader/patch webpack-hot-middleware/client?path=http://localhost:3000/__webpack_hmr&reload=true ./app/index
```

--- a/src/reconciler/proxies.js
+++ b/src/reconciler/proxies.js
@@ -3,8 +3,7 @@ import { resetClassProxies } from '../proxy/createClassProxy';
 import { isCompositeComponent, isReactClass } from '../internal/reactUtils';
 import configuration from '../configuration';
 import { incrementHotGeneration } from '../global/generation';
-
-const merge = require('lodash/merge');
+import { merge } from './utils';
 
 let signatures;
 let proxiesByID;

--- a/src/reconciler/utils.js
+++ b/src/reconciler/utils.js
@@ -74,3 +74,25 @@ export const areSwappable = (a, b) => {
   }
   return false;
 };
+
+export function merge(...sources) {
+  let acc = {};
+  for (const source of sources) {
+    if (source instanceof Array) {
+      if (!(acc instanceof Array)) {
+        acc = [];
+      }
+      acc = [...acc, ...source];
+    } else if (source instanceof Object) {
+      for (const entry of Object.entries(source)) {
+        const key = entry[0];
+        let value = entry[1];
+        if (value instanceof Object && key in acc) {
+          value = merge(acc[key], value);
+        }
+        acc = { ...acc, [key]: value };
+      }
+    }
+  }
+  return acc;
+}

